portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,19 +29,25 @@
 import javax.servlet.http.HttpServletResponse;
 import com.google.gson.Gson;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles sending and receiving comments.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private static final Query QUERY = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
   private static final DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
   private static final Gson GSON = new Gson();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    PreparedQuery results = DATASTORE.prepare(QUERY);
-    
+    // Get filter parameters
+    int maxComments = getNumComments(request);
+
+    Query query = makeQueryFromParams(request);
```

Comment:
Could you prepare a query that limits reading maxComments from Datastore vs doing it in memory? Former will be efficient.